RecoTauTag/RecoTau/plugins/RecoTauPiZeroStripPlugin3.cc

```diff
@@ -0,0 +1,400 @@
+
+#include "RecoTauTag/RecoTau/interface/RecoTauPiZeroPlugins.h"
+
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/TauReco/interface/RecoTauPiZero.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "CommonTools/CandUtils/interface/AddFourMomenta.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include "RecoTauTag/RecoTau/interface/RecoTauCommonUtilities.h"
+#include "RecoTauTag/RecoTau/interface/RecoTauQualityCuts.h"
+#include "RecoTauTag/RecoTau/interface/RecoTauVertexAssociator.h"
+#include "RecoTauTag/RecoTau/interface/CombinatoricGenerator.h"
+
+//-------------------------------------------------------------------------------
+// CV: the following headers are needed only for debug print-out
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+//-------------------------------------------------------------------------------
+
+#include "TString.h"
+#include "TFormula.h"
+
+namespace reco { namespace tau {
+
+namespace {
+  // Apply a hypothesis on the mass of the strips.
+  math::XYZTLorentzVector applyMassConstraint(
+      const math::XYZTLorentzVector& vec,double mass) {
+    double factor = sqrt(vec.energy()*vec.energy()-mass*mass)/vec.P();
+    return math::XYZTLorentzVector(
+        vec.px()*factor,vec.py()*factor,vec.pz()*factor,vec.energy());
+  }
+}
+
+class RecoTauPiZeroStripPlugin3 : public RecoTauPiZeroBuilderPlugin 
+{
+ public:
+  explicit RecoTauPiZeroStripPlugin3(const edm::ParameterSet&, edm::ConsumesCollector &&iC);
+  virtual ~RecoTauPiZeroStripPlugin3();
+  // Return type is auto_ptr<PiZeroVector>
+  return_type operator()(const reco::PFJet&) const override;
+  // Hook to update PV information
+  virtual void beginEvent() override;
+  
+ private:
+  typedef std::vector<reco::PFCandidatePtr> PFCandPtrs;
+  void addCandsToStrip(RecoTauPiZero&, PFCandPtrs&, const std::vector<bool>&, std::set<size_t>&, bool&) const;
+
+  RecoTauVertexAssociator vertexAssociator_;
+
+  RecoTauQualityCuts* qcuts_;
+  bool applyElecTrackQcuts_;
+  double minGammaEtStripSeed_;
+  double minGammaEtStripAdd_;
+
+  double minStripEt_;
+
+  std::vector<int> inputPdgIds_;  // type of candidates to clusterize
+  TFormula* etaAssociationDistance_; // size of strip clustering window in eta direction
+  TFormula* phiAssociationDistance_; // size of strip clustering window in phi direction
+
+  bool updateStripAfterEachDaughter_;
+  int maxStripBuildIterations_;
+
+  // Parameters for build strip combinations
+  bool combineStrips_;
+  int maxStrips_;
+  double combinatoricStripMassHypo_;
+
+  AddFourMomenta p4Builder_;
+
+  int verbosity_;
+};
+
+namespace
+{
+  TFormula* makeFunction(const std::string& functionName, const edm::ParameterSet& pset)
```

Comment:
it looks like in your code you call only Eval, which is a const function.
change the return type to ```std::unique_ptr< const TFormula> ```